docs/contributing/code-review.md

```diff
@@ -35,6 +35,28 @@ Here are some guidelines for merging and reviewing in this case:
   - Updating the user image of a hub.
   :::
 
+## Changing authentication related configuration
+
+Extra caution must be taken when changing authentication related configuration. Since the current testing infrastructure doesn't automatically test the authentication workflow, any issue within the authentication process might go undetected and cause an outage, if no manual testing is conducted before deploying it to all the hubs.
+
+To deploy changes to the authentication workflow, follow these steps:
+1. **Propose changes**. Change the codebase and open a PR.
+1. **Manually deploy changes to a few kinds of staging hubs**. The hubs use different authentication mechanisms, based on their needs, but testing on the staging hubs listed below should cover all the possible types. The hubs are:
+   - cluster: `leap`, hub: `staging` (GitHub)
+   - cluster: `utoronto`, hub: `staging` (Azure AD)
```

Comment:
Wondering if we should have additional demos for all the auth options so we do not depend on community-specific hubs... but maybe I am overthinking this one 😉 .